src/app/auth/authentication.service.ts

```diff
@@ -135,27 +142,31 @@ export class AuthenticationService {
 
   refreshToken() {
     if (this.isLoggedIn()) {
-      let headers = new Headers({ 'Content-Type': 'application/json' });
-      let options: RequestOptions = new RequestOptions({ headers: headers });
       let refreshTokenUrl = this.apiUrl + 'token/refresh';
       let refreshToken = localStorage.getItem('refresh_token');
       let body = JSON.stringify({ 'refresh_token': refreshToken });
-      this.http.post(refreshTokenUrl, body, options)
-        .map((response: Response) => {
-          let responseJson = response.json();
-          let token = this.processTokenResponse(responseJson.token);
-          this.clearTimeoutId = null;
-          this.setupRefreshTimer(token.expires_in);
-          return token;
+      const httpOptions = {
+        headers: new HttpHeaders({
+          'Content-Type':  'application/json'
         })
-        .catch(response => {
-          // Additionally catch a 400 from keycloak
-          if (response.status === 400 || isAuthenticationError(response)) {
-            this.broadcaster.broadcast('authenticationError', response);
-          }
-          return Observable.of({} as Token);
-        })
-        .subscribe(token => {
+      };
+      this.http.post<any>(refreshTokenUrl, body, httpOptions)
+        .pipe(
+          map((t: any) => {
+            let token = this.processTokenResponse(t.token);
+            this.clearTimeoutId = null;
+            this.setupRefreshTimer(token.expires_in);
+            return token;
+          }),
+          catchError(response => {
+            // Additionally catch a 400 from keycloak
+            if (response.status === 400 || isAuthenticationError(response)) {
```

Comment:
I don't think the call to`isAuthenticationError` needed here still if we have the interceptor.

Reply:
I'm unsure about this one. It triggers the broadcast if it is there. As that was added to deal with something, I would almost rather leave it alone and refactor after.

Reply:
I think we can remove these explicit calls to `isAuthenticationError` since we have interceptors for this in place now.